modules/drivers/athena/src/metabase/driver/athena.clj

```diff
@@ -142,9 +142,14 @@
 
 ;;; ------------------------------------------------- date functions -------------------------------------------------
 
+(def ^:dynamic *loading-data*
+  "HACK! Whether we're loading data (e.g. in [[metabase.test.data.athena]]). We can't use `timestamp with time zone`
+  literals when loading data because Athena doesn't let you use a `timestamp with time zone` value for a `timestamp`
+  column, and you can only have `timestamp` columns when actually creating them."
+  false)
```

Comment:
To fix the problems we had in CI I manually wiped all the data and recreated everything, and then ran into a bug with the data-loading code: it was trying to insert `timestamp with time zone` literals into `timezone` columns, which Athena does not allow, since you cannot create a `timestamp with time zone` column. Thus when loading data we need to compile `OffsetDateTime`s and `ZonedDateTime`s to `timestamp` literals. 

I had actually written both versions of the code for `OffsetDateTime` and `ZonedDateTime` previously -- one that compiled it to a `timestamp with time zone` literal (which is fine in `SELECT` queries), and another to a `timestamp` literal. So now that this flag was in place we can just choose the correct behavior based on what we're doing.